scripts/world.gd

```diff
@@ -79,6 +92,15 @@ func get_closest_available_health(from_position: Vector3) -> HealthPickup:
 	var pickups = []
 	pickups.append_array(health_pickups)
 
+	# Filter returns the filtered array, but sort is in-place
 	pickups.sort_custom(func(a, b): return from_position.distance_to(a.global_transform.origin) < from_position.distance_to(b.global_transform.origin))
-	pickups.filter(func(a): return a.health_is_available)
+	pickups = pickups.filter(func(a): return a.health_is_available)
 	return pickups.front()
+
+func get_closest_available_weapon(from_position: Vector3) -> Weapon:
+	var weapon_array = []
+	weapon_array.append_array(weapons)
+
+	weapon_array.sort_custom(func(a, b): return from_position.distance_to(a.global_transform.origin) < from_position.distance_to(b.global_transform.origin))
+	weapon_array = weapon_array.filter(func(a): return !a.is_held)
+	return weapon_array.front()
```

Comment:
The front() array method will throw an error if the array is empty. Let's first check if the array contains any values and return null if not.
```
E 0:00:07:0268   world.gd:106 @ get_closest_available_weapon(): Can't take value from empty array.
  <C++ Error>    Condition "_p->array.size() == 0" is true. Returning: Variant()
  <C++ Source>   core/variant/array.cpp:319 @ front()
  <Stack Trace>  world.gd:106 @ get_closest_available_weapon()
                 find_weapon_state.gd:11 @ execute()
                 state_machine.gd:19 @ _process()
                 ai_actor_controller.gd:14 @ _process()
```

Reply:
I've added ternary checks for this and the other two object finding methods in World.